Implement --sdnotify cmdline option to control sd-notify behavior (#6693)
containers/libpod: docs/source/markdown/podman-create.1.md

This is useful to run a container without requiring any image management, the rootfs
of the container is assumed to be managed externally.

**--sdnotify**=**container**|**conmon-only**|**ignore**

Determines how to use the NOTIFY_SOCKET, as passed with systemd and Type=notify.

Default is **container**, which means allow the OCI runtime to proxy the socket into the
container to receive ready notification.  Podman will set the MAINPID to conmon's pid.
The **conmon-only** option sets MAINPID to conmon's pid, and sends READY when the container
has started.  The socket is never passed to the runtime or the container.
The **ignore** option removes NOTIFY_SOCKET from the environment for itself and child processes,

> for the case where some other service above us uses NOTIFY_SOCKET but we shouldn't mess with it.

Reviewer: ```suggestion
for the case where some other service above Podman uses NOTIFY_SOCKET and Podman should not revise it.
```

Author: revised
